src/iidm/converter/xml/NetworkXml.cpp

```diff
@@ -217,11 +220,21 @@ Network NetworkXml::read(const std::string& /*filename*/, std::istream& is, cons
         throw powsybl::xml::XmlStreamException(err.what());
     }
 
-    // TODO(sebalaig) handle anonymization by reading csv file if it exists
-    // For now on, use a FakeAnonymizer
-    FakeAnonymizer anonymizer;
+    boost::filesystem::path csvPath = boost::filesystem::path(filename).replace_extension("csv");
+    std::unique_ptr<Anonymizer> anonymizer;
+
+    if (boost::filesystem::exists(csvPath)) {
+        anonymizer = stdcxx::make_unique<SimpleAnonymizer>();
+        boost::filesystem::ifstream csvStream(csvPath);
+        if (!csvStream.is_open()) {
+            throw PowsyblException(stdcxx::format("Unable to open file '%1%' for reading", csvPath.filename().string()));
+        }
+        anonymizer->read(csvStream);
+    } else {
+        anonymizer = stdcxx::make_unique<FakeAnonymizer>();
+    }
 
-    NetworkXmlReaderContext context(anonymizer, reader, options, version);
+    NetworkXmlReaderContext context(*anonymizer, reader, options, version);
```

Comment:
To be discussed: I'm not sure this is correct to get the pointer like this. How is it converted to a reference? As the anonymizer is now always created in the implementation and not exposed in the API, should we move the instance to make context the new owner. I think the lifetime of the context and the anonymizer are the same